runtime/doc/develop.txt

```diff
@@ -424,6 +424,11 @@ The basic steps to make changes to the code:
 For any non-trivial change, please always create a pull request on github,
 since this triggers the test suite.
 
+Ideally, one PR (pull request) only included one commit; or if it contained
+many changes, then you maybe better split it to some commits; however, if some
+commits was going to modify same file, then it's recommended to squash them to
+one commit, to reduce conflict and be easy to check.
```

Comment:
I don't agree here. I think this is better:

> A PR should ideally contain a single commit for a single logical change. However, you can include several commits if you want to group multiple logical, atomic changes in one PR. This can also make longer PRs easier to review. Be sure to describe the reasoning for your changes in each commit message, as this greatly helps with the review process. In cases where each commit handles different logical changes, they will also be applied as separate patches in Vim’s repository.

Reply:
in that case if different logical changes, i think it should be done in different PRs, and in that case, you will find github would notice you later change on same file would be marked as 'out date change', then that PR author knew  to update/rebase his PR.

Reply:
yes, it is ok to all belong to the same PR too.
my main point was: better be not commit several commits on same fs in one PR, to those commits better be squash to signal commit.